accumulator/batchproof.go

```diff
@@ -223,25 +231,24 @@ func verifyBatchProof(
 }
 
 // Reconstruct takes a number of leaves and rows, and turns a block proof back
-// into a partial proof tree.  Should leave bp intact
+// into a partial proof tree. Should leave bp intact
 func (bp *BatchProof) Reconstruct(
 	numleaves uint64, forestRows uint8) (map[uint64]Hash, error) {
 
 	if verbose {
 		fmt.Printf("reconstruct blockproof %d tgts %d hashes nl %d fr %d\n",
 			len(bp.Targets), len(bp.Proof), numleaves, forestRows)
 	}
-	proofCopy := make([]Hash, len(bp.Proof))
-	copy(proofCopy, bp.Proof)
 	proofTree := make(map[uint64]Hash)
 
+	// If there is nothing to reconstruct, return empty map
 	if len(bp.Targets) == 0 {
 		return proofTree, nil
 	}
+	proofCopy := bp.Proof // back up proof
```

Comment:
nit: maybe call this just `proof`, `targets` is also not called `targetsCopy`.

Reply:
Sure. This really shouldn't even be a method though if there's gonna be a copy... Something to do later down the road.